samples/snippets/src/main/java/IdTokenFromImpersonatedCredentials.java

```diff
@@ -0,0 +1,159 @@
+    // depending on the level of access you need.
+    // Example: The following scope lets you view and manage Pub/Sub topics and subscriptions.
+    // For more information, see: https://developers.google.com/identity/protocols/oauth2/scopes
+    String scope = "https://www.googleapis.com/auth/pubsub";
+
+    // The service name for which the id token is requested. Service name refers to the
+    // logical identifier of an API service, such as "pubsub.googleapis.com".
+    String targetAudience = "pubsub.googleapis.com";
+
+    // The service account name of the limited-privilege account for whom the credential is created.
+    String impersonatedServiceAccount = "name@project.service.gserviceaccount.com";
+
+    getIdTokenFromImpersonatedCredentials(
+        projectId, jsonCredentialPath, impersonatedServiceAccount, scope, targetAudience);
+  }
+
+  public static void getIdTokenFromImpersonatedCredentials(
+      String projectId,
+      String jsonCredentialPath,
+      String impersonatedServiceAccount,
+      String scope,
+      String targetAudience)
+      throws GeneralSecurityException, IOException {
+
+    // Initialize the IAMCredentials service with the source credential and scope.
+    IAMCredentials service = null;
+    try {
+      service = initService(jsonCredentialPath, scope);
+    } catch (IOException | GeneralSecurityException e) {
+      System.out.println("Unable to initialize service: \n" + e);
+      return;
+    }
+
+    // delegates: The chained list of delegates required to grant the final accessToken.
+    //
+    // If set, the sequence of identities must have "Service Account Token Creator" capability
+    // granted to the preceding identity.
+    // For example, if set to [serviceAccountB, serviceAccountC], the source credential must have
+    // the Token Creator role on serviceAccountB. serviceAccountB must have the Token Creator on
+    // serviceAccountC. Finally, C must have Token Creator on impersonatedServiceAccount.
+    //
+    // If left unset, source credential must have that role on impersonatedServiceAccount.
+    List<String> delegates = null;
+
+    // Set the target audience and Token options.
+    GenerateIdTokenRequest idTokenRequest =
+        new GenerateIdTokenRequest()
+            .setAudience(targetAudience)
+            .setDelegates(delegates)
+            // Setting this will include email in the id token.
+            .setIncludeEmail(Boolean.TRUE);
+
+    // Generate the id token for the impersonated service account, using the generateIdToken()
+    // from IAMCredentials class.
+    GenerateIdToken idToken =
+        service
+            .projects()
+            .serviceAccounts()
+            .generateIdToken(
+                String.format(
+                    "projects/%s/serviceAccounts/%s", projectId, impersonatedServiceAccount),
+                idTokenRequest);
+
+    // Verify the obtained id token. This is done at the receiving end of the OIDC endpoint.
```

Comment:
I won't be using this the way the page is currently structured. Please exclude when you add region tags.